qf_lib/data_providers/db_connection_providers.py

```diff
@@ -0,0 +1,45 @@
+#     Copyright 2016-present CERN – European Organization for Nuclear Research
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+import logging
+from abc import ABCMeta, abstractmethod
+
+from blpapi import Session
+from sqlalchemy.exc import DatabaseError
+from sqlalchemy.orm import sessionmaker
+
+from settings import Settings
+
+
+class DBConnectionProvider(metaclass=ABCMeta):
+    """Factory class which produces database Connection objects."""
+
+    def __init__(self, settings: Settings, **kwargs):
+        self.engine = self._create_engine(settings, **kwargs)
+        self.Session = sessionmaker(bind=self.engine)
+        self.session = None
```

Comment:
let's rename the `self.Session` to something like `self.session_maker`, it's a bit confusing having both `self.session and self.Session`